### PR TITLE
feat: created heap_sort_2.c

### PR DESCRIPTION
**Pull request by DhruvPasricha**
#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: sorting/heap_sort_2.c [hunk 1]
```diff
@@ -0,0 +1,138 @@
+/**
+ * @file
+ * @author [Dhruv Pasricha](https://github.com/DhruvPasricha)
+ * @brief [Heap Sort](https://en.wikipedia.org/wiki/Heapsort)
```
Comment (Panquesito7) [inline sorting/heap_sort_2.c:4]:
```suggestion
 * @brief [Heap Sort](https://en.wikipedia.org/wiki/Heapsort) implementation
```

File: sorting/heap_sort_2.c [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+ * @returns 0 on exit
+ */
+int main()
+{
+    /* Intializes random number generator */
+    srand(time(NULL));
+
+    test(); // run self-test implementations
+    return 0;
+}
```
Comment (Panquesito7) [inline sorting/heap_sort_2.c:138]:
```suggestion
}

```

**Review by Panquesito7 — COMMENTED** [review]
👍 Well done!

**Comment by DhruvPasricha** [comment]
> 👍 Well done!

Thank You! 😄

File: sorting/heap_sort_2.c [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+void heapSort(int8_t *arr, const uint8_t size)
+{
+    if (size <= 1)
+    {
+        return;
+    }
+
+    for (uint8_t i = 0; i < size; i++)
+    {
+        /*Pushing arr[i] to the heap*/
```
Comment (Panquesito7) [inline sorting/heap_sort_2.c:87]:
```suggestion
        // Pushing `arr[i]` to the heap
```

File: sorting/heap_sort_2.c [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+        /*heapifyUp Adjusts arr[i] to the correct position in the heap*/
+        heapifyUp(arr, i);
+    }
+
+    for (uint8_t i = size - 1; i >= 1; i--)
+    {
+        /*Moving current root to the end*/
```
Comment (Panquesito7) [inline sorting/heap_sort_2.c:95]:
```suggestion
        // Moving current root to the end
```

File: sorting/heap_sort_2.c [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+        swap(&arr[0], &arr[i]);
+
+        /*heapifyDown Adjusts new root to the correct position in the heap*/
```
Comment (Panquesito7) [inline sorting/heap_sort_2.c:98]:
```suggestion
        // `heapifyDown` adjusts new root to the correct position in the heap
```

File: sorting/heap_sort_2.c [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+    free(arr);
+}
+
+/**
+ * @brief Main function
+ * @returns 0 on exit
+ */
+int main()
+{
+    /* Intializes random number generator */
```
Comment (Panquesito7) [inline sorting/heap_sort_2.c:133]:
```suggestion
    // Intializes random number generator
```

**Review by Panquesito7 — COMMENTED** [review]
Please enable [GitHub Actions](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions) in your repository of this fork in this link: https://github.com/DhruvPasricha/C/actions

File: sorting/heap_sort_2.c [hunk 7]
```diff
@@ -0,0 +1,138 @@
+/**
+ * @file
+ * @author [Dhruv Pasricha](https://github.com/DhruvPasricha)
+ * @brief [Heap Sort](https://en.wikipedia.org/wiki/Heapsort) implementation
```
Comment (Panquesito7) [inline sorting/heap_sort_2.c:4]: Please add a detailed description of the algorithm.

File: sorting/heap_sort_2.c [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+
+    }
+}
+
+/**
+ * @brief Self-test implementations
+ * @returns void
+ */
+static void
+test()
```
Comment (Panquesito7) [inline sorting/heap_sort_2.c:109]:
```suggestion
static void test()
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
> Please enable [GitHub Actions](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions) in your repository of this fork in this link: https://github.com/DhruvPasricha/C/actions

This **is** required. Please follow the steps above.

File: sorting/heap_sort_2.c (hunk 7 above)
Comment (DhruvPasricha) [reply sorting/heap_sort_2.c:4]: Okay

**Comment by DhruvPasricha** [comment]
> > Please enable [GitHub Actions](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions) in your repository of this fork in this link: https://github.com/DhruvPasricha/C/actions
> 
> This **is** required. Please follow the steps above.

Sir, I have already enabled them. Please check.

File: sorting/heap_sort_2.c (hunk 7 above)
Comment (DhruvPasricha) [reply sorting/heap_sort_2.c:4]: Okay Sir

File: sorting/heap_sort_2.c [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+ *  Heap-sort can be thought of as an improved selection sort:
+ *  like selection sort, heap sort divides its input into a sorted
+ *  and an unsorted region, and it iteratively shrinks the unsorted
+ *  region by extracting the largest element from it and inserting
+ *  it into the sorted region. Unlike selection sort,
+ *  heap sort does not waste time with a linear-time scan of the
+ *  unsorted region; rather, heap sort maintains the unsorted region
+ *  in a heap data structure to more quickly find the largest element
+ *  in each step.
+ *  Time Complexity : O(Nlog(N))
```
Comment (Panquesito7) [inline sorting/heap_sort_2.c:16]:
```suggestion
 * Heap-sort is a comparison-based sorting algorithm.
 * Heap-sort can be thought of as an improved selection sort:
 * like selection sort, heap sort divides its input into a sorted
 * and an unsorted region, and it iteratively shrinks the unsorted
 * region by extracting the largest element from it and inserting
 * it into the sorted region. Unlike selection sort,
 * heap sort does not waste time with a linear-time scan of the
 * unsorted region; rather, heap sort maintains the unsorted region
 * in a heap data structure to more quickly find the largest element
 * in each step.
 * Time Complexity : O(Nlog(N))
```

File: sorting/heap_sort_2.c [hunk 10]
```diff
@@ -0,0 +1,156 @@
...
+ * @file
+ * @author [Dhruv Pasricha](https://github.com/DhruvPasricha)
+ * @brief [Heap Sort](https://en.wikipedia.org/wiki/Heapsort) implementation
+ * @details
+ * Heap-sort is a comparison-based sorting algorithm.
+ * Heap-sort can be thought of as an improved selection sort:
+ * like selection sort, heap sort divides its input into a sorted
+ * and an unsorted region, and it iteratively shrinks the unsorted
+ * region by extracting the largest element from it and inserting
+ * it into the sorted region. Unlike selection sort,
```
Comment (Panquesito7) [inline sorting/heap_sort_2.c:11]:
```suggestion
 * it into the sorted region.
 *
 * Unlike selection sort,
```

File: sorting/heap_sort_2.c [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+/**
+ * @brief heapifyDown Adjusts new root to the correct position in the heap
+ * This heapify procedure can be thought of as building a heap from
+ * the top down by successively shifting downward to establish the
+ * heap property.
+ * @param arr array to be sorted
+ * @param size size of array
+ * @return void
+*/
+
```
Comment (Panquesito7) [inline sorting/heap_sort_2.c:49]:
```suggestion
```

File: sorting/heap_sort_2.c [hunk 12]
```diff
@@ -0,0 +1,158 @@
...
+ * it into the sorted region.
+ *
+ * Unlike selection sort,
+ * heap sort does not waste time with a linear-time scan of the
+ * unsorted region; rather, heap sort maintains the unsorted region
+ * in a heap data structure to more quickly find the largest element
+ * in each step.
+ * Time Complexity : O(Nlog(N))
+ */
+
```
Comment (Panquesito7) [inline sorting/heap_sort_2.c:20]:
```suggestion
```

**Review by Panquesito7 — APPROVED** [review]
Amazing work @DhruvPasricha, I like how you made the code very nicely! 🎉
Code and documentation are pretty good and refined; LGTM. 👍 😄

**Comment by DhruvPasricha** [comment]
> Amazing work @DhruvPasricha, I like how you made the code very nicely! 🎉
> Code and documentation are pretty good and refined; LGTM. 👍 😄

Thank you so much!  😄😄

**Review by ayaankhan98 — APPROVED** [review]
Great work @DhruvPasricha and @Panquesito7

**Comment by DhruvPasricha** [comment]
> Great work @DhruvPasricha and @Panquesito7

Thank you so much! 😄😄

**Pull request merged**